destination/destination_integration_test.go

```diff
@@ -0,0 +1,223 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package destination
+
+import (
+	"context"
+	"testing"
+
+	config "github.com/conduitio-labs/conduit-connector-nats-jetstream/config"
+	test "github.com/conduitio-labs/conduit-connector-nats-jetstream/test"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/matryer/is"
+)
+
+func TestDestination_Open(t *testing.T) {
+	t.Parallel()
+
+	is := is.New(t)
+
+	conn, err := test.GetTestConnection()
+	is.NoErr(err)
+
+	err = test.CreateTestStream(conn, t.Name(), []string{
+		"foo_destination",
+	})
+	is.NoErr(err)
+
+	t.Run("success, jetstream", func(t *testing.T) {
```

Comment:
Personally, whenever I see multiple `t.Run()` calls, I think if it makes sense having a table driven test, or just having separate test methods.

Reply:
I've split it into separate test functions in the https://github.com/conduitio-labs/conduit-connector-nats-jetstream/pull/10/commits/5d190f808b484d02f22e7294ea29801227c75c1b